tools/src/bin/docgen/main.rs

```diff
@@ -202,31 +202,33 @@ fn comment_and_requirement(
     src: &[u8],
 ) -> Result<(Option<Comment>, Option<Feature>), Box<dyn Error>> {
     let prev_prev = prev.prev_named_sibling();
-    let mut maybe_comment = Comment::new(prev, src).ok();
 
-    // If node wasn't a comment, see if it was an expression_statement
-    // that itself was preceded by a comment.  This skips over
-    // expression-like preprocessor attributes on function decls.
-    if let (None, "expression_statement", Some(prev_prev)) =
-        (&maybe_comment, prev.kind(), prev_prev)
-    {
-        maybe_comment = Comment::new(prev_prev, src).ok();
+    // In the simple case, `prev` is a comment and `prev_prev` may
+    // be a feature requirement.
+    if let Ok(comment) = Comment::new(prev, src) {
```

Comment:
This commit is still a little confusing to me, it seems to have a bunch of things going on that interact, making it hard to understand what the logical changes are. Suggest at least splitting out a commit that only does the `maybe_comment.is_none()` inversion thing?

Reply:
Split up further.